cpp/src/copying/slice.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 BlazingDB, Inc.
+ *     Copyright 2019 Christian Noboa Mardini <christian@blazingdb.com>
+ *     Copyright 2019 William Scott Malpica <william@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/cudf.h>
+#include <utilities/error_utils.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/copying.hpp>
+#include <algorithm>
+
+namespace cudf {
+
+namespace experimental {
+
+namespace detail {
+
+std::vector<cudf::column_view> slice(cudf::column_view const& input,
```

Comment:
This doesn't need a `detail` API because it doesn't allocate any memory nor execute any kernels, which means it doesn't need a stream.

Reply:
Wouldn't it be good to have them in detail and having stream if we are planning any changes in future which might require these changes.

Reply:
I think it would be better if such changes were necessary to add these APIs at that time. Otherwise it could cause confusion as to why the detail API needs to exist.